### PR TITLE
Search counter

### PR DESCRIPTION
**Pull request by ejohnson44**
Got the search counter up and running - also includes Reid's fixes to some crashes

### CODE REVIEW

File: SoundStream/project.properties [hunk 1]
```diff
@@ -12,4 +12,5 @@
 
 # Project target.
 target=android-17
-android.library.reference.1=../third_party/SlidingMenuLibrary
+android.library.reference.1=..\\third_party\\SlidingMenuLibrary
+android.library.reference.2=../../../../AppData/Local/Android/android-sdk/extras/android/support/v7/appcompat
```
Comment (dgreenhalgh) [inline SoundStream/project.properties]: This came up in another pull request, but is this filepath Windows-specific?

File: SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java [hunk 2]
```diff
@@ -63,6 +65,8 @@
     
     private MusicAdapter mMusicAdapter;
     
+    private View headerView;
```
Comment (dgreenhalgh) [inline SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java]: Nitpick, but a better name would be mHeaderView to reinforce the concept that this is a member variable

File: SoundStream/project.properties (hunk 1 above)
Comment (ejohnson44) [inline SoundStream/project.properties]: It definitely is, but I don't remember how to not include it but not have it disappear from my project.

Comment (reidbaker) [inline SoundStream/project.properties]: git rm should remove it from git but not remove it locally.

**Comment by dgreenhalgh** [comment]
:+1:

File: SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java [hunk 3]
```diff
@@ -313,13 +350,13 @@ public void onClick(View v) {
         @Deprecated
         private void updateMusicFromLibrary() {
             this.updateMusic(getMusicLibraryFromService());
-            notifyDataSetChanged();
+//            notifyDataSetChanged();
```
Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java]: Remove?

File: SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java [hunk 4]
```diff
@@ -313,13 +350,13 @@ public void onClick(View v) {
...
         }
         
         private void updateMusicFromQuery(String query) {
             if (query != null) {
                 this.updateMusic(getMusicLibraryFromQuery(query));
-                notifyDataSetChanged();    
+//                notifyDataSetChanged();
```
Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java]: Remove?

File: SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java (hunk 3 above)
Comment (theJenix) [inline SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java]: Why was this removed?  Is this handled in the class updateMusic function?

File: SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java (hunk 4 above)
Comment (theJenix) [inline SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java]: Same comment as above.

File: SoundStream/src/com/thelastcrusade/soundstream/CoreActivity.java [hunk 5]
```diff
@@ -188,7 +190,14 @@ public boolean onCreateOptionsMenu(Menu menu) {
         
         if (searchView != null) {
             searchView.setSearchableInfo(searchManager.getSearchableInfo(new ComponentName(this, SearchActivity.class)));
-            searchView.setIconifiedByDefault(true);
+            searchView.setIconified(true);
+            searchView.setOnClickListener(new OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    v.setFocusable(true);
+                    v.requestFocusFromTouch();
```
Comment (theJenix) [inline SoundStream/src/com/thelastcrusade/soundstream/CoreActivity.java:198]: It seems like this shouldn't be necessary (i.e. using a click listener to change the focusability of a view and to request focus should be handled by the core, no?), but I presume it is for some reason.  Can you add a comment explaining what's going on here?

File: SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java [hunk 6]
```diff
@@ -137,9 +141,11 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         
         super.onCreateView(inflater, container, savedInstanceState);
         View v = inflater.inflate(R.layout.list, container, false);
+ 
+        setListAdapter(null);
```
Comment (theJenix) [inline SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java:146]: What's going on with this line?  Are we trying to reuse functionality that isn't exactly right for this purpose, and would it be better to refactor to support this case?

File: SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java [hunk 7]
```diff
@@ -163,12 +190,22 @@ public void onServiceBound() {
...
     @Override
     public void onStart() {
         super.onStart();
 //        ((CoreActivity)getActivity()).getTracker().sendView(TAG);
     }
+    
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        if(mHeaderView != null){
```
Comment (theJenix) [inline SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java:193]: How can mHeaderView be null?  Can you please add a comment?

File: SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java [hunk 8]
```diff
@@ -147,10 +153,31 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+                mHeaderView = inflater.inflate(R.layout.search_counter,null);
+                mHeaderView.setLayoutParams(new ListView.LayoutParams(ListView.LayoutParams.MATCH_PARENT, ListView.LayoutParams.WRAP_CONTENT));
+                
+
+                mMusicAdapter.registerDataSetObserver(new DataSetObserver() {
+                    @Override
+                    public void onChanged() {
+                      super.onChanged();
+                      TextView resultsCounter = (TextView)mHeaderView.findViewById(R.id.results_count);
+//                      Log.i(TAG, "Music Adapter Count on observed change" + mMusicAdapter.getCount());
```
Comment (theJenix) [inline SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java]: You can wrap these with Log.isLoggable(TAG, Log.INFO) and leave these log statements in place, and we can enable that level in logcat if we need that level of detail.  Also, if appropriate, you LogUtil.isLogAvailable to make this test junit testable (if it's more appropriate to use the integration tests, I don't think we need to check.  This is just for straight up unit tests).

File: SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java [hunk 9]
```diff
@@ -147,10 +153,31 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+                    }
+                });
+                
+            
             } else {
                 Log.w(TAG, "Fragment recieved arguments but no query");
             }
         }
+        else{
+            mQuery = null;
```
Comment (theJenix) [inline SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java:174]: This is a small nit, but an alternative to using null would be to use the empty string (or an appropriately named constant), or some other string to indicate all results.  This would clean up the test for nulls, and may streamline the actual search (i.e. if all queries were valid, including "no query", you wouldn't need any special cases for no query).  This is not a showstopper, more just a suggestion for any future changes.

File: SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java (hunk 3 above)
Comment (ejohnson44) [inline SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java]: Yeah, it's done in another function so it's not necessary here. I'll get rid of the commented bit

File: SoundStream/src/com/thelastcrusade/soundstream/CoreActivity.java (hunk 5 above)
Comment (ejohnson44) [inline SoundStream/src/com/thelastcrusade/soundstream/CoreActivity.java:198]: This is some of what Reid did, so I'll let him tell me what comment should go there

File: SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java (hunk 6 above)
Comment (ejohnson44) [inline SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java:146]: It's some of what I had to do to make this 2.3 compatible. If we actually change away from that then I can get rid of this

File: SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java (hunk 8 above)
Comment (ejohnson44) [inline SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java]: That was just for my own experimentation I think - it's not something that is necessary

File: SoundStream/src/com/thelastcrusade/soundstream/CoreActivity.java (hunk 5 above)
Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/CoreActivity.java:198]: this makes it so when you tap on the search icon in addition to opening the text box it selects it.

File: SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java (hunk 9 above)
Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/components/MusicLibraryFragment.java:174]: Like that solution

File: SoundStream/src/com/thelastcrusade/soundstream/CoreActivity.java (hunk 5 above)
Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/CoreActivity.java:198]: //Make search box selected after tapping search icon

**Comment by reidbaker** [comment]
Is this good to pull in?

**Comment by reidbaker** [comment]
@ejohnson44 says this can be pulled in
